docs/technical_overview.rst

```diff
@@ -0,0 +1,165 @@
+CommCare Technical Overview
```

Comment:
I don't actually think this is a good example of documentation that should live in the repo. This is much more of a public-facing website-content that is more likely to get updated by communications or marketing people that it is to get updated by developers. One criteria that makes sense (to me at least) for what belongs here is - "would I intuitively go edit this thing when I make a change in the code?" This one doesn't seem to fit the bill.